user_tasks/tasks.py

```diff
@@ -77,17 +77,22 @@ def status(self):
         """
         task_id = self.request.id
         try:
+            # Most calls are for existing objects, don't waste time
+            # preparing creation arguments unless necessary
             return UserTaskStatus.objects.get(task_id=task_id)
         except UserTaskStatus.DoesNotExist:
             # Probably an eager task that skipped the before_task_publish
-            # signal.  Create a record for it.
+            # signal (or an atomic view where the new record hasn't been
+            # committed yet).  Create a record for it.
             arguments_dict = self.arguments_as_dict(*self.request.args, **self.request.kwargs)
             name = self.generate_name(arguments_dict)
             task_class = '.'.join([self.__class__.__module__, self.__class__.__name__])
             total_steps = self.calculate_total_steps(arguments_dict)
             user_id = arguments_dict['user_id']
-            return UserTaskStatus.objects.create(
-                name=name, task_id=task_id, task_class=task_class, total_steps=total_steps, user_id=user_id)
+            # Use get_or_create() again just in case another process created it in the meantime
+            return UserTaskStatus.objects.get_or_create(
```

Comment:
Those are only ever called from the `before_task_published` signal handler, so I don't think there's any chance of that.  Also, a couple of them are generating random task_id field values because there isn't a real corresponding Celery task (we just want a logical parent of the whole chain or chord), so get_or_create wouldn't work properly on those (no unique-indexed field by which to look up an existing record).